lib/matplotlib/tests/test_bezier.py

```diff
@@ -0,0 +1,35 @@
+from matplotlib import bezier
+
+import numpy as np
+import scipy.integrate
+
+
+test_curves = [
+    # cubic, degenerate derivative
+    bezier.BezierSegment([[0, 0], [1, 0], [1, 1], [0, 1]]),
+    # cubic, could be quadratic
+    bezier.BezierSegment([[0, 0], [1, 1/2], [1, 1/2], [0, 1]]),
+    # interior extrema determine extents
+    bezier.BezierSegment([[0, 0], [4, 0], [-4, 1], [0, 1]]),
+    # a quadratic curve
+    bezier.BezierSegment([[0, 0], [0, 1], [1, 1]]),
+    # a linear curve
+    bezier.BezierSegment([[0, 1], [1, 1]]),
+    # a point
+    bezier.BezierSegment([[1, 2]])
+]
+
+
+def _integral_arc_area(B):
+    """(Signed) area swept out by ray from origin to curve."""
+    dB = B.differentiate(B)
+    def integrand(t):
+        dr = dB(t).T
+        n = np.array([dr[1], -dr[0]])
+        return (B(t).T @ n) / 2
+    return scipy.integrate.quad(integrand, 0, 1)[0]
```

Comment:
Testing our area formula (`bezier.BezierSegment.arc_area`) is most naturally done by comparing to the naive integral implementation. However, this required me to add `scipy` to the testing dependencies.

I already test this formula (indirectly) in `test_path.test_signed_area_curve` and `test_path.test_signed_area_unit_rectangle`, so if we prefer to not add any dependency on scipy at all, I can simply remove this test.

Reply:
It was agreed on a call back in June that it's okay to add this Scipy dependency as long as it's just an optional testing dependency.